xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-api/src/main/java/org/xwiki/livedata/LiveDataEntryDescriptor.java

```diff
@@ -49,4 +52,36 @@ public void setIdProperty(String idProperty)
     {
         this.idProperty = idProperty;
     }
+
+    /**
+     * @since 17.4.0RC1
+     */
+    @Override
+    public boolean equals(Object o)
+    {
+        if (this == o) {
+            return true;
+        }
+
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+
+        LiveDataEntryDescriptor that = (LiveDataEntryDescriptor) o;
+
+        return new EqualsBuilder()
```

Comment:
Not sure it worth creating a EqualsBuilder for a single property. `Objects.equals` should do fine.

Reply:
This special case is not documented on https://dev.xwiki.org/xwiki/bin/view/Community/CodeStyle/JavaCodeStyle/#HEquals2FHashCodeandToStringimplementations, so I aimed for code consistency.